# SCVdb å¤„ç†è¿‡ç¨‹

## 1. æ”¶é›† scATAC å’Œ variant æ•°æ®

### 1.1 æ”¶é›† scATAC æ•°æ®

> åœ¨ **scATACRef** æ”¶é›†

### 1.2 æ”¶é›† **variant** æ•°æ®

> æ”¶é›†çš„æ•°æ®ä¿¡æ¯

| èµ„æº                                | æ¥æº                             | PMID     | æ€»å…±æ€§çŠ¶æ•°é‡ | ç­›é€‰åæ€§çŠ¶æ•°é‡ | æ•°æ®é“¾æ¥                                                                | genome |
|-----------------------------------|--------------------------------|----------|--------|---------|---------------------------------------------------------------------|--------|
| GWAS catalog fine-mapping results | Annalisa Buniello et al., 2019 | 30445434 | 381    | 3       | https://www.ebi.ac.uk/gwas/downloads/summary-statistics             | hg38   |
| UKBB fine-mapping results         | Kanai, M. et al., 2021         |          | 94     | 94      | https://www.finucanelab.org/data                                    | hg19   |
| FinnGen fine-mapping results      | Kanai, M. et al., 2022         | 36653562 | 86     | 51      | https://www.finngen.fi/en/access_results                            | hg38   |
| BBJ fine-mapping results          | Kanai, M. et al., 2021         |          | 79     | 77      | https://humandbs.biosciencedbc.jp/en/hum0197-v15#hum0197.v5.gwas.v1 | hg19   |
| CAUSALdb fine-mapping results     | Jianhua Wang et al., 2019      | 31691819 | 3292   | 2144    | http://www.mulinlab.org/causaldb/index.html                         | hg19   |

> çªå˜æ•°æ®è½¬åŒ–ä¸ºæŒ‡å®šæ ¼å¼ä¿¡æ¯
>
> åœ¨è½¬åŒ–è¿‡ç¨‹ä¸­é‡åˆ°å¤šç§æ–¹æ³•è·å– PIP ä¸ªå€¼çš„æ—¶å€™ï¼Œåªé€‰å– fine-mapping å¾—åˆ°çš„ PIP å€¼

> æ¯åˆ—çš„å«ä¹‰:  `chr	postion	postion	variant_name	PIP trait`
>
> ä¾‹å¦‚:

```bed
chr11	305619	305619	rs6421984	0.106024	baso
chr11	306920	306920	rs7115703	0.108784	baso
chr11	307244	307244	rs10751648	0.211543	baso
chr11	307696	307696	rs7480524	0.131486	baso
```

> ä¹‹åçªå˜æ•°æ®é€šè¿‡ liftOver è¿›è¡Œè½¬åŒ–ä¸º hg38

## 2. å¯¹çªå˜æ•°æ®å½¢æˆæ•°æ®åº“è¡¨æ ¼å¤„ç†

> çªå˜æ•°æ®çš„æ–‡ä»¶åæ˜¯å­˜å‚¨æ•°æ®è¡¨æ ¼ä¸­çš„ id åŠ ä¸Šå‚è€ƒåŸºå› ç»„

## 3. å¯¹ Fragments æ–‡ä»¶è¿›è¡Œ ArchR å¤„ç†

> å¾—åˆ° scATAC æ•°æ®è¡¨è¾¾æ•°æ®

## 4. å°† scATAC å’Œçªå˜æ•°æ®æ•´åˆ

> ä½¿ç”¨ SCAVENGE æ–¹æ³•è¿›è¡Œæ•´åˆ

## 5. è·¯å¾„å­˜å‚¨è¯´æ˜

> `path` ä¸ºæ ¹è·¯å¾„

- ğŸ“`{path}`
  - ğŸ“database: å­˜å‚¨ `SCVdb` æ•°æ®åº“æ•°æ®
    - ğŸ“code: å­˜å‚¨è·‘æ•°æ®åº“æ•°æ®çš„ä»£ç 
      - ğŸ“`{server}`: ä»£è¡¨ä¸åŒçš„æœåŠ¡å™¨, å­˜å‚¨æ‰§è¡Œ g-chromVAR å’Œ SCAVENGE æ–¹æ³•çš„ä»£ç 
        - ğŸ“„library.R: å¼•ç”¨çš„ R åŒ…
        - ğŸ“„static_function.R: å•ç»†èƒå’Œæ•´åˆä»£ç çš„å¤„ç†
        - ğŸ“„integration.R: å°†å¤šä¸ªæ€§çŠ¶æˆ–ç–¾ç—…å¾ªç¯è·‘æ•°æ®
        - ğŸ“„run.R: æ‰§è¡Œçš„æ€»ä»£ç 
    - ğŸ“sc_variant: å­˜å‚¨æ®åº“æ•°æ®è¾“å‡ºçš„æ ¹è·¯å¾„
      - ğŸ“result: å­˜å‚¨ g-chromVAR å’Œ SCAVENGE æ–¹æ³•ç»“æœ
        - ğŸ“`{scATAC-seq}`: ä»£è¡¨ä¸åŒçš„å•ç»†èƒæ•°æ®, å­˜å‚¨æŸä¸ªå•ç»†èƒæ•°æ®æ•´åˆæ‰€æœ‰çš„æ€§çŠ¶æˆ–ç–¾ç—…çš„ç»“æœæ•°æ®
          - ğŸ“„`{scATAC-seq}__{genome}__{trait_label_file}__mat.txt`: å­˜å‚¨ TRS, å¯Œé›†çŠ¶æ€ç­‰ä¿¡æ¯æ–‡ä»¶
          - ğŸ“„`{scATAC-seq}__{genome}__{trait_label_file}__mat_info.rda`: å­˜å‚¨äº†ç»“æœ txt å’Œ mutual-KNN ä¿¡æ¯æ–‡ä»¶
      - ğŸ“scATAC: å­˜å‚¨äº†å•ç»†èƒè·‘ä¹‹å‰çš„å¤„ç†æ–‡ä»¶ä»¥åŠè·‘ä¹‹åç»“æœæ–‡ä»¶
        - ğŸ“`{scATAC-seq}`: ä»£è¡¨ä¸åŒçš„å•ç»†èƒæ•°æ®, å­˜å‚¨æŸä¸ªå•ç»†èƒæ•°æ®è·‘ä¹‹å‰çš„å¤„ç†æ–‡ä»¶ä»¥åŠè·‘ä¹‹åç»“æœæ–‡ä»¶
          - ğŸ“„`{scATAC-seq}.txt`: ç»†èƒæ³¨é‡Šæ–‡ä»¶
          - ğŸ“„`{scATAC-seq}_all.rda`: å•ç»†èƒå¤„ç†æ–‡ä»¶, åœ¨ä¸æ€§çŠ¶æˆ–è€…ç–¾ç—…è¿›è¡Œæ•´åˆè¯»å–çš„æ˜¯è¿™ä¸ªæ–‡ä»¶
          - ğŸ“„`{scATAC-seq}_SE_gvar_SE_gvar_bg.rda`: å•ç»†èƒå¤„ç†çš„ä¸­é—´æ–‡ä»¶, å†…å®¹æ¯” ${scATAC-seq}_all.rda è¿™ä¸ªæ–‡ä»¶å°‘
          - ğŸ“„`{scATAC-seq}_trs_scavenge_data.h5ad`: æŸä¸ªå•ç»†èƒæ•°æ®æ•´åˆæ‰€æœ‰çš„æ€§çŠ¶æˆ–ç–¾ç—…çš„ç»“æœæ•°æ®å½¢æˆä¸€ä¸ª h5ad æ–‡ä»¶æ ¼å¼
      - ğŸ“table: å­˜å‚¨ä¸æ•°æ®åº“æ­å»ºç›¸å…³çš„æ•°æ®
        - ğŸ“download: å­˜å‚¨é€šè¿‡ SnapATAC2 å¤„ç†åçš„ scATAC-seq æ•°æ®
          - ğŸ“scatac: å­˜å‚¨é€šè¿‡ SnapATAC2 å¤„ç†åçš„ scATAC-seq æ•°æ®
            - ğŸ“„`{scATAC-seq}_snapATAC2.h5ad`: åœ¨ç½‘ç«™ä¸Šä¸‹è½½é¡µä¸‹è½½çš„æ•°æ®
          - ğŸ“„cp.sh: å°†é€šè¿‡ SnapATAC2 å¤„ç†åçš„ scATAC-seq æ•°æ®éƒ½å¤åˆ¶åˆ°é€šè·¯å¾„ä¸‹çš„ scatac ä¸­
        - ğŸ“homer: å­˜å‚¨ HOMER è·‘å‡ºæ¥çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„è½¬å½•å› å­æ•°æ®
          - ğŸ“hg19: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„è½¬å½•å› å­æ•°æ®
            - ğŸ“„`t_homer_{group}`.txt:  ä¸æ•°æ®åº“ä¸­ `t_homer_hg19_{group}` è¡¨å¯¹åº”
          - ğŸ“hg38: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„è½¬å½•å› å­æ•°æ®
            - ğŸ“„`t_homer_{group}`.txt:  ä¸æ•°æ®åº“ä¸­ `t_homer_hg38_{group}` è¡¨å¯¹åº”
          - ğŸ“„t_homer_tf_trait_count.txt: å­˜å‚¨åœ¨ä¸åŒå‚è€ƒåŸºå› ç»„ä¸‹ TF åœ¨æ€§çŠ¶æˆ–ç–¾ç—…æ ·æœ¬ä¸è½¬å½•å› å­çš„æ•°é‡, ä¸æ•°æ®åº“ä¸­ `t_tf_trait_count` è¡¨å¯¹åº”
          - ğŸ“„t_homer_trait_tf.txt: å­˜å‚¨åœ¨ä¸åŒå‚è€ƒåŸºå› ç»„ä¸‹æ€§çŠ¶æˆ–ç–¾ç—…æ ·æœ¬ä¸è½¬å½•å› å­çš„æ˜ å°„å…³ç³»
          - ğŸ“„t_homer_trait_tf_hg19.txt: å­˜å‚¨ hg19 æ€§çŠ¶æˆ–ç–¾ç—…æ ·æœ¬ä¸è½¬å½•å› å­çš„æ˜ å°„å…³ç³», ä¸æ•°æ®åº“ä¸­ `t_trait_tf_hg19` è¡¨å¯¹åº”
          - ğŸ“„t_homer_trait_tf_hg38.txt: å­˜å‚¨ hg38 æ€§çŠ¶æˆ–ç–¾ç—…æ ·æœ¬ä¸è½¬å½•å› å­çš„æ˜ å°„å…³ç³», ä¸æ•°æ®åº“ä¸­ `t_trait_tf_hg38` è¡¨å¯¹åº”
          - ğŸ“„trait_tf_hg19.txt: è¿™ä¸ªæ–‡ä»¶æ˜¯å¯¹ `hg19` æ–‡ä»¶æ•°æ®æ”¾åœ¨äº†ä¸€èµ· (æ²¡æœ‰ç”¨åˆ°, ä¿ç•™)
          - ğŸ“„trait_tf_hg38.txt: è¿™ä¸ªæ–‡ä»¶æ˜¯å¯¹ `hg38` æ–‡ä»¶æ•°æ®æ”¾åœ¨äº†ä¸€èµ· (æ²¡æœ‰ç”¨åˆ°, ä¿ç•™)
        - ğŸ“magma: å­˜å‚¨ MAGMA è·‘å‡ºæ¥çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› æ•°æ®
          - ğŸ“gene_enrichment_trait: å­˜å‚¨æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› çš„ GO, KEGG ç­‰åŸºå› å¯Œé›†åŠŸèƒ½
            - ğŸ“hg19: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› çš„ GO, KEGG ç­‰åŸºå› å¯Œé›†åŠŸèƒ½
            - ğŸ“hg38: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› çš„ GO, KEGG ç­‰åŸºå› å¯Œé›†åŠŸèƒ½
          - ğŸ“gene_enrichment_trait_table: å­˜å‚¨æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› çš„ GO, KEGG ç­‰åŸºå› å¯Œé›†åŠŸèƒ½
            - ğŸ“hg19: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå›  GO, KEGG ç­‰åŸºå› å¯Œé›†åŠŸèƒ½
              - ğŸ“„`t_gene_enrichment_trait_hg19_{group}.txt`: ä¸æ•°æ®åº“ä¸­ `t_gene_enrichment_trait_hg19_{group}` è¡¨å¯¹åº”
            - ğŸ“hg38: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå›  GO, KEGG ç­‰åŸºå› å¯Œé›†åŠŸèƒ½
              - ğŸ“„`t_gene_enrichment_trait_hg38_{group}.txt`: ä¸æ•°æ®åº“ä¸­ `t_gene_enrichment_trait_hg38_{group}` è¡¨å¯¹åº”
          - ğŸ“hg19: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› æ•°æ® (å¸¦æœ‰å¯Œé›†çš„å¾—åˆ†)
            - ğŸ“„`t_magma_{group}.txt`: ä¸æ•°æ®åº“ä¸­ `t_magma_hg19_{group}` è¡¨å¯¹åº”
          - ğŸ“hg19_anno: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› æ•°æ® (å¸¦æœ‰çªå˜å’ŒåŸºå› æ˜ å°„)
            - ğŸ“„`t_magma_{group}.txt`: ä¸æ•°æ®åº“ä¸­ `t_magma_anno_hg19_{group}` è¡¨å¯¹åº”
          - ğŸ“hg38: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› æ•°æ® (å¸¦æœ‰å¯Œé›†çš„å¾—åˆ†)
            - ğŸ“„`t_magma_{group}.txt`: ä¸æ•°æ®åº“ä¸­ `t_magma_hg38_{group}` è¡¨å¯¹åº”
          - ğŸ“hg38_anno: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› æ•°æ® (å¸¦æœ‰çªå˜å’ŒåŸºå› æ˜ å°„)
            - ğŸ“„`t_magma_{group}.txt`: ä¸æ•°æ®åº“ä¸­ `t_magma_anno_hg38_{group}` è¡¨å¯¹åº”
          - ğŸ“„t_magma_gene_trait_count.txt: å­˜å‚¨åœ¨ä¸åŒå‚è€ƒåŸºå› ç»„ä¸‹ Gene åœ¨æ€§çŠ¶æˆ–ç–¾ç—…æ ·æœ¬ä¸è½¬å½•å› å­çš„æ•°é‡, ä¸æ•°æ®åº“ä¸­ `t_gene_trait_count` è¡¨å¯¹åº”
          - ğŸ“„t_magma.txt: å­˜å‚¨åœ¨ä¸åŒå‚è€ƒåŸºå› ç»„ä¸‹æ€§çŠ¶æˆ–ç–¾ç—…æ ·æœ¬ä¸åŸºå› çš„æ˜ å°„å…³ç³»
          - ğŸ“„t_magma_hg19.txt: å­˜å‚¨ hg19 æ€§çŠ¶æˆ–ç–¾ç—…æ ·æœ¬ä¸åŸºå› çš„æ˜ å°„å…³ç³», ä¸æ•°æ®åº“ä¸­ `t_trait_gene_hg19` è¡¨å¯¹åº”
          - ğŸ“„t_magma_hg38.txt: å­˜å‚¨ hg38 æ€§çŠ¶æˆ–ç–¾ç—…æ ·æœ¬ä¸åŸºå› çš„æ˜ å°„å…³ç³», ä¸æ•°æ®åº“ä¸­ `t_trait_gene_hg38` è¡¨å¯¹åº”
        - ğŸ“scatac: å­˜å‚¨ä¸ scATAC-seq æ•°æ®ç›¸å…³çš„æ•°æ®åº“å†…å®¹
          - ğŸ“difference_gene: å•ç»†èƒæ ·æœ¬çš„å·®å¼‚åŸºå› å¯Œé›†
            - `{scATAC-seq}_difference_gene_data.txt`: æŸä¸ªå•ç»†èƒæ ·æœ¬çš„å·®å¼‚åŸºå› å¯Œé›† (æ²¡æœ‰ç”¨åˆ°, ä¿ç•™)
          - ğŸ“difference_tf: å•ç»†èƒæ ·æœ¬çš„å·®å¼‚è½¬å½•å› å­å¯Œé›†
            - `{scATAC-seq}_difference_tf_data.txt`: æŸä¸ªå•ç»†èƒæ ·æœ¬çš„å·®å¼‚è½¬å½•å› å­å¯Œé›† (æ²¡æœ‰ç”¨åˆ°, ä¿ç•™)
          - ğŸ“gene_enrichment_table: å•ç»†èƒæ ·æœ¬çš„å·®å¼‚åŸºå›  GO, KEGG ç­‰é€šè·¯å¯Œé›†
            - `{scATAC-seq}_gene_enrichment_data.txt`: æŸä¸ªå•ç»†èƒæ ·æœ¬çš„å·®å¼‚åŸºå›  GO, KEGG ç­‰é€šè·¯å¯Œé›†, ä¸æ•°æ®åº“ä¸­ `t_gene_enrichment_{scATAC-seq}` è¡¨å¯¹åº”
          - ğŸ“„difference_gene_data.txt: æ‰€æœ‰å•ç»†èƒæ ·æœ¬å·®å¼‚åŸºå› å¯Œé›†, ä¸æ•°æ®åº“ä¸­ `t_difference_gene` è¡¨å¯¹åº”
          - ğŸ“„difference_tf_data.txt: æ‰€æœ‰å•ç»†èƒæ ·æœ¬å·®å¼‚è½¬å½•å› å­å¯Œé›†, ä¸æ•°æ®åº“ä¸­ `t_difference_tf` è¡¨å¯¹åº”
          - ğŸ“„gene_enrichment_table_data.txt: æ‰€æœ‰å•ç»†èƒæ ·æœ¬çš„å·®å¼‚åŸºå›  GO, KEGG ç­‰é€šè·¯å¯Œé›† (æ²¡æœ‰ç”¨åˆ°, ä¿ç•™)
          - ğŸ“„sample_gene_data.txt: å­˜å‚¨å•ç»†èƒæ ·æœ¬ä¸åŸºå› çš„æ˜ å°„å…³ç³»æ–‡ä»¶, ä¸æ•°æ®åº“ä¸­ `t_sample_gene` è¡¨å¯¹åº”
        - ğŸ“trait_variant_overlap: å­˜å‚¨æ‰€æœ‰å•ç»†èƒå’Œæ‰€æœ‰æ€§çŠ¶æˆ–ç–¾ç—…æ˜¯å¦å­˜åœ¨ overlap çš„æƒ…å†µ (æ˜¯å¦ overlap: !æ˜¯å¦ TRS å¾—åˆ†å…¨ä¸ºé›¶)
          - ğŸ“„gchromvar_sample_enrichment.txt: g-chromVAR æ–¹æ³• overlap æƒ…å†µä»¥ txt æ–‡ä»¶å­˜å‚¨, ä¸æ•°æ®åº“ä¸­ `t_difference_tf` è¡¨å¯¹åº”
          - ğŸ“„scavenge_sample_enrichment.txt: SCAVENGE æ–¹æ³• overlap æƒ…å†µä»¥ txt æ–‡ä»¶å­˜å‚¨, ä¸æ•°æ®åº“ä¸­ `t_difference_tf` è¡¨å¯¹åº”
          - ğŸ“„trs_overlap_gchromvar.h5ad: g-chromVAR æ–¹æ³• overlap æƒ…å†µä»¥ h5ad æ–‡ä»¶å­˜å‚¨ (æ²¡æœ‰ç”¨åˆ°, ä¿ç•™)
          - ğŸ“„trs_overlap_scavenge.h5ad: SCAVENGE æ–¹æ³• overlap æƒ…å†µä»¥ h5ad æ–‡ä»¶å­˜å‚¨ (æ²¡æœ‰ç”¨åˆ°, ä¿ç•™)
        - ğŸ“trs: å­˜å‚¨ç½‘ç«™è¯»å–æ–‡ä»¶å±•ç¤ºå¯è§†åŒ–çš„æ•°æ®å†…å®¹ (è¿™é‡Œå¯¹ç»“æœ h5ad æ–‡ä»¶è¿›è¡Œåˆ†æ–‡ä»¶ 100 ä»½)
          - ğŸ“`{scATAC-seq}`: ä»£è¡¨ä¸åŒçš„å•ç»†èƒæ•°æ®, å­˜å‚¨æŸä¸ªå•ç»†èƒæ•°æ®ä¸åˆ†ç»„æ€§çŠ¶æˆ–è€…ç–¾ç—…çš„ç»“æœ
            - ğŸ“cell_type: ä»¥ç»†èƒç±»å‹ä¸ºåˆ†è¾¨ç‡çš„ TRS ç»“æœ
              - ğŸ“„`{scATAC-seq}_cell_type_trs_gchromvar.h5ad`: g-chromVAR æ–¹æ³•è®¡ç®—å‡ºæ¥çš„ cell types-traits çŸ©é˜µ
              - ğŸ“„`{scATAC-seq}_cell_type_trs_scavenge.h5ad`: SCAVENGE æ–¹æ³•è®¡ç®—å‡ºæ¥çš„ cell types-traits çŸ©é˜µ
            - ğŸ“chunk: TRS ç»“æœ
              - ğŸ“„`{scATAC-seq}_{group}_trs_gchromvar.h5ad`: g-chromVAR æ–¹æ³•è®¡ç®—å‡ºæ¥çš„ cells-traits çŸ©é˜µ
              - ğŸ“„`{scATAC-seq}_{group}_trs_scavenge.h5ad`: SCAVENGE æ–¹æ³•è®¡ç®—å‡ºæ¥çš„ cells-traits çŸ©é˜µ
        - ğŸ“trs_big: å­˜å‚¨ç½‘ç«™ä¸‹è½½çš„æ–‡ä»¶ (è¿™é‡Œå­˜å‚¨ç»“æœ h5ad æ–‡ä»¶, æœªåˆ†è¡¨)
          - ğŸ“`{scATAC-seq}`: ä»£è¡¨ä¸åŒçš„å•ç»†èƒæ•°æ®, å­˜å‚¨æŸä¸ªå•ç»†èƒæ•°æ®ä¸æ‰€æœ‰æ€§çŠ¶æˆ–è€…ç–¾ç—…çš„ç»“æœ
              - ğŸ“„`{scATAC-seq}_trs_gchromvar.h5ad`: g-chromVAR æ–¹æ³•è®¡ç®—å‡ºæ¥çš„ cells-traits çŸ©é˜µ
              - ğŸ“„`{scATAC-seq}_trs_scavenge.h5ad`: SCAVENGE æ–¹æ³•è®¡ç®—å‡ºæ¥çš„ cells-traits çŸ©é˜µ
        - ğŸ“variant: å­˜å‚¨ç½‘ç«™çªå˜ä¿¡æ¯æ•°æ®åº“çš„æ•°æ®
          - ğŸ“hg19: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„ fine-mapping ç»“æœçš„æ•°æ®
            - ğŸ“„`t_variant_{group}_hg19.txt`: ä¸æ•°æ®åº“ä¸­ `t_variant_hg19_{group}` è¡¨å¯¹åº”
          - ğŸ“hg38: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„ fine-mapping ç»“æœçš„æ•°æ®
            - ğŸ“„`t_variant_{group}_hg38.txt`: ä¸æ•°æ®åº“ä¸­ `t_variant_hg38_{group}` è¡¨å¯¹åº”
          - ğŸ“„t_trait_chr_count.txt: å­˜å‚¨æ€§çŠ¶æˆ–è€…ç–¾ç—…ä¸­ä¸åŒæŸ“è‰²è´¨çš„æ•°é‡, ä¸æ•°æ®åº“ä¸­ `t_trait_chr_count` è¡¨å¯¹åº”
      - ğŸ“variant: å­˜å‚¨è·‘æ•°æ®çš„è¾“å…¥ fine-mapping ç»“æœçš„æ•°æ®
        - ğŸ“hg19: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„ fine-mapping ç»“æœçš„æ•°æ®
        - ğŸ“hg38: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„ fine-mapping ç»“æœçš„æ•°æ®
        - ğŸ“„fine_mapping_result.pkl: ç”¨æ–‡ä»¶å­˜å‚¨æ‰€æœ‰å‚è€ƒåŸºå› ç»„çš„ fine-mapping ç»“æœçš„æ•°æ®
        - ğŸ“„fine_mapping_result_hg19.pkl: ç”¨æ–‡ä»¶å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„ fine-mapping ç»“æœçš„æ•°æ®
        - ğŸ“„fine_mapping_result_hg38.pkl: ç”¨æ–‡ä»¶å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„ fine-mapping ç»“æœçš„æ•°æ®
  - ğŸ“gene: å­˜å‚¨åŸºå› ç›¸å…³çš„æ•°æ®
    - ğŸ“download: ç°åœ¨ä¸ SnapATAC2 ä¸€è‡´çš„åŸºå› ç‰ˆæœ¬
      - ğŸ“„gencode.v41.annotation.gtf.gz: å‚è€ƒåŸºå› ç»„ä¸º hg38
      - ğŸ“„gencode.v41lift37.annotation.gtf.gz: å‚è€ƒåŸºå› ç»„ä¸º hg19
    - ğŸ“result: é€šè¿‡ download æ•°æ®è¿›è¡Œå¤„ç†åçš„è§„èŒƒè¡¨æ ¼æ•°æ®
      - ğŸ“„gene_hg19_all.txt: å½¢æˆè¡¨æ ¼çš„æ•°æ®, å«æœ‰æ‰€æœ‰åˆ—
      - ğŸ“„gene_hg38_all.txt: å½¢æˆè¡¨æ ¼çš„æ•°æ®, å«æœ‰æ‰€æœ‰åˆ—
      - ğŸ“„gene_data.txt: å½¢æˆè¾“å…¥æ•°æ®åº“ä¸­çš„æ•°æ®, ä¸æ•°æ®åº“ä¸­ `t_gene` è¡¨å¯¹åº”
      - ğŸ“„gene_hg19_data.txt: ä» gene_data.txt è¡¨æŠ½å– hg19 æ•°æ®
      - ğŸ“„gene_hg38_data.txt: ä» gene_data.txt è¡¨æŠ½å– hg38 æ•°æ®
    - ğŸ“annotation: å¸¦æœ‰åŸºå› æ³¨é‡Šçš„ç›¸å…³æ–‡ä»¶
      - ğŸ“dbSNP: åœ¨ dbSNP ä¸‹è½½çš„æ–‡ä»¶
        - ğŸ“common_snp_chunk:
          - ğŸ“hg19:
          - ğŸ“hg38:
        - ğŸ“„dbsnp_common_snp_hg19.txt:
        - ğŸ“„dbsnp_common_snp_hg19.cvf:
        - ğŸ“„dbsnp_common_snp_hg38.txt:
        - ğŸ“„dbsnp_common_snp_hg38.cvf:
      - ğŸ“dbSUPER: åœ¨ dbSUPER ä¸‹è½½çš„æ–‡ä»¶
        - ğŸ“all_hg19_bed:
        - ğŸ“liftOver:
        - ğŸ“„all_hg19_bed.zip:
        - ğŸ“„dbsuper_super_enhancer_hg19.txt:
        - ğŸ“„dbsuper_super_enhancer_hg38.txt:
      - ğŸ“GTEx: åœ¨ GTEx ä¸‹è½½çš„æ–‡ä»¶
        - ğŸ“eqtl_chunk:
          - ğŸ“hg19:
          - ğŸ“hg38:
        - ğŸ“GTEx_Analysis_v10_eQTL_updated:
        - ğŸ“liftOver:
          - ğŸ“input: è½¬æˆå¯¹åº”å‚è€ƒåŸºå› ç»„çš„è¾“å…¥æ–‡ä»¶
          - ğŸ“output: è½¬æˆå¯¹åº”å‚è€ƒåŸºå› ç»„çš„è¾“å‡ºæ–‡ä»¶
          - ğŸ“result: æœ€ç»ˆå¯¹åº”å‚è€ƒåŸºå› ç»„çš„åŸºå› æ³¨é‡Šæ–‡ä»¶
        - ğŸ“„gtex_v10_eqtl_hg19.txt:
        - ğŸ“„gtex_v10_eqtl_hg38.txt:
        - ğŸ“„gtex_v10_eqtl_hg38.tar:
      - ğŸ“gwasATLAS: åœ¨ gwasATLAS ä¸‹è½½çš„æ–‡ä»¶
        - ğŸ“liftOver:
          - ğŸ“input: è½¬æˆå¯¹åº”å‚è€ƒåŸºå› ç»„çš„è¾“å…¥æ–‡ä»¶
          - ğŸ“output: è½¬æˆå¯¹åº”å‚è€ƒåŸºå› ç»„çš„è¾“å‡ºæ–‡ä»¶
          - ğŸ“result: æœ€ç»ˆå¯¹åº”å‚è€ƒåŸºå› ç»„çš„åŸºå› æ³¨é‡Šæ–‡ä»¶
        - ğŸ“„gwasATLAS_v20191115.txt:
        - ğŸ“„gwasATLAS_v20191115_riskloci.txt:
        - ğŸ“„gwasatlas_v20191115_risk_snp_hg19.txt:
        - ğŸ“„gwasatlas_v20191115_risk_snp_hg38.txt:
      - ğŸ“SEA: åœ¨ SEA ä¸‹è½½çš„æ–‡ä»¶
        - ğŸ“liftOver:
          - ğŸ“input: è½¬æˆå¯¹åº”å‚è€ƒåŸºå› ç»„çš„è¾“å…¥æ–‡ä»¶
          - ğŸ“output: è½¬æˆå¯¹åº”å‚è€ƒåŸºå› ç»„çš„è¾“å‡ºæ–‡ä»¶
          - ğŸ“result: æœ€ç»ˆå¯¹åº”å‚è€ƒåŸºå› ç»„çš„åŸºå› æ³¨é‡Šæ–‡ä»¶
        - ğŸ“„sea_v3_enhancer_hg19.txt:
        - ğŸ“„sea_v3_enhancer_hg38.txt:
        - ğŸ“„sea_v3_super_enhancer_hg19.txt:
        - ğŸ“„sea_v3_super_enhancer_hg38.txt:
        - ğŸ“„sea_v3_super_enhancer_hg38.bed:
      - ğŸ“SEdb: åœ¨ SEdb ä¸‹è½½çš„æ–‡ä»¶
        - ğŸ“enhancer_chunk:
        - ğŸ“liftOver:
          - ğŸ“input: è½¬æˆå¯¹åº”å‚è€ƒåŸºå› ç»„çš„è¾“å…¥æ–‡ä»¶
          - ğŸ“output: è½¬æˆå¯¹åº”å‚è€ƒåŸºå› ç»„çš„è¾“å‡ºæ–‡ä»¶
          - ğŸ“result: æœ€ç»ˆå¯¹åº”å‚è€ƒåŸºå› ç»„çš„åŸºå› æ³¨é‡Šæ–‡ä»¶
        - ğŸ“„SEdb_2.0_sample_information.txt:
        - ğŸ“„sedb_v2_enhancer_hg19.txt:
        - ğŸ“„sedb_v2_enhancer_hg38.txt:
        - ğŸ“„sedb_v2_enhancer_hg38_middle.txt:
        - ğŸ“„sedb_v2_super_enhancer_hg19.txt:
        - ğŸ“„sedb_v2_super_enhancer_hg38.txt:
        - ğŸ“„sedb_v2_super_enhancer_hg38_middle.txt:
        - ğŸ“„SE_package_hg38.bed:
        - ğŸ“„TE_package_hg38.bed:
  - ğŸ“project_code: å­˜å‚¨ä»£ç 
    - ğŸ“scvdb_reproducibility: æ­¤æ–‡ä»¶å¤¹è¡¨ç¤ºæœ¬é¡¹ç›®çš„æ ¹è·¯å¾„
  - ğŸ“scATAC: 
    - ğŸ“`{GSE_ID}`: å­˜å‚¨ä¸åŒå•ç»†èƒæ ·æœ¬æ•´ä½“çš„å¤„ç†æµç¨‹æ•°æ®
      - ğŸ“data: å­˜å‚¨é€šè¿‡æ¥æº rds æ–‡ä»¶å¤„ç†æˆç»Ÿä¸€è§„æ ¼çš„æ•°æ®
        - `{scATAC-seq}`: å­˜å‚¨ä¸åŒå•ç»†èƒæ ·æœ¬å¤„ç†æˆç»Ÿä¸€è§„æ ¼çš„æ•°æ®
          - ğŸ“meta: å­˜å‚¨å•ç»†èƒæ ·æœ¬çš„å…ƒæ•°æ®
            - ğŸ“„barcodes.tsv: barcodes ä¿¡æ¯
            - ğŸ“„matrix.mtx: ç¨€ç–çŸ©é˜µä¿¡æ¯
            - ğŸ“„peaks.bed: peak ä¿¡æ¯
          - ğŸ“„annotation.txt: æœªå¤„ç†å‰æ‰€æœ‰ç»†èƒæ³¨é‡Šçš„æ–‡ä»¶
          - ğŸ“„annotation_stdn.txt: æœªå¤„ç†å•ç»†èƒæ³¨é‡Šçš„ç»Ÿä¸€è§„èŒƒæ–‡ä»¶
          - ğŸ“„`{scATAC-seq}_sc_atac.h5ad`: æœªé¢„å¤„ç†çš„å•ç»†èƒæ ·æœ¬æ–‡ä»¶
          - ğŸ“„`{scATAC-seq}_fragments.tsv.gz`: å•ç»†èƒæ ·æœ¬çš„ fragments æ–‡ä»¶
          - ğŸ“„`{scATAC-seq}_sc_atac_snapATAC2.h5ad`: é¢„å¤„ç†åçš„å•ç»†èƒæ ·æœ¬æ–‡ä»¶
          - ğŸ“„`{scATAC-seq}_cell_anno.txt`: å•ç»†èƒé¢„å¤„ç†åç»†èƒæ³¨é‡Šçš„æ–‡ä»¶
          - ğŸ“„`{scATAC-seq}_cell_anno_stdn.txt`: å•ç»†èƒé¢„å¤„ç†åç»†èƒæ³¨é‡Šçš„ç»Ÿä¸€è§„èŒƒæ–‡ä»¶ (æœ€ç»ˆ)
          - ğŸ“„`{scATAC-seq}_cell_type_stdn.txt`: å•ç»†èƒé¢„å¤„ç†åç»†èƒç±»å‹æ³¨é‡Šçš„ç»Ÿä¸€è§„èŒƒæ–‡ä»¶
          - ğŸ“„`{scATAC-seq}_gene_expression_data.h5ad`: å•ç»†èƒæ ·æœ¬çš„åŸºå› è¡¨è¾¾æ•°æ®
          - ğŸ“„`{scATAC-seq}_difference_gene.h5ad`: å•ç»†èƒæ ·æœ¬çš„å·®å¼‚åŸºå› æ•°æ®
          - ğŸ“„`{scATAC-seq}_gene_enrichment.txt`: å•ç»†èƒæ ·æœ¬çš„å·®å¼‚åŸºå› å¯Œé›†æ–‡ä»¶
          - ğŸ“„`{scATAC-seq}_difference_peak.h5ad`: å•ç»†èƒæ ·æœ¬çš„å·®å¼‚ peak æ•°æ®
          - ğŸ“„`{scATAC-seq}_tf_activity_data.h5ad`: å•ç»†èƒæ ·æœ¬çš„å·®å¼‚è½¬å½•å› å­æ–‡ä»¶
      - ğŸ“„`{scATAC-seq}_ATAC.rds`: ä¸åŒå•ç»†èƒæ ·æœ¬çš„æ¥æº rds æ–‡ä»¶
      - ğŸ“„`{scATAC-seq}_metadata.txt`: ä¸åŒå•ç»†èƒæ ·æœ¬çš„æ¥æºçš„æ³¨é‡Šæ–‡ä»¶
  - ğŸ“topic: SCIV ç®—æ³•çš„æ ¹è·¯å¾„ä¿¡æ¯å†…å®¹
  - ğŸ“variant: å…³äºæ‰€æœ‰ fine-mapping ç»“æœæ–‡ä»¶çš„å­˜å‚¨ (æ¥æºåˆ°æœ€ç»ˆ)
    - ğŸ“source: æ‰€æœ‰ fine-mapping ç»“æœæ–‡ä»¶çš„æ¥æºæ•°æ®
      - ğŸ“BBJ: æ‰€æœ‰ BBJ é˜Ÿåˆ— fine-mapping ç»“æœæ–‡ä»¶çš„æ¥æºæ•°æ®
        - ğŸ“decompression: å¯¹æ‰€æœ‰ä¸‹è½½çš„æ–‡ä»¶è¿›è¡Œå‹ç¼©
          - ğŸ“„`BBJ.{trait}.Kanai2021.FINEMAP.tsv.gz`: FINEMAP è½¯ä»¶è·‘å‡ºæ¥çš„ç»“æœ
          - ğŸ“„`BBJ.{trait}.Kanai2021.SuSiE.tsv.gz`: SuSiEè½¯ä»¶è·‘å‡ºæ¥çš„ç»“æœ (æœªä½¿ç”¨)
        - ğŸ“download: ä¸‹è½½ BBJ çš„æ‰€æœ‰ fine-mapping ç»“æœæ–‡ä»¶
          - ğŸ“„`hum0197.v5.finemap.{trait}.v1.zip`: ä¸‹è½½çš„ BBJ çš„æ€§çŠ¶æˆ–ç–¾ç—…æ–‡ä»¶
        - ğŸ“finemap: å¯¹æ‰€æœ‰ FINEMAP è½¯ä»¶è·‘å‡ºæ¥çš„ç»“æœè¿›è¡Œå‹ç¼©
          - ğŸ“„`BBJ.{trait}.Kanai2021.FINEMAP.tsv.txt`: å¯¹ FINEMAP æ–¹æ³•å¾—åˆ°çš„ç»“æœè§£å‹åçš„æ–‡ä»¶
      - ğŸ“CAUSALdb: æ‰€æœ‰ CAUSALdb é˜Ÿåˆ— fine-mapping ç»“æœæ–‡ä»¶çš„æ¥æºæ•°æ®
        - ğŸ“download: ä¸‹è½½ fine-mapping ç»“æœæ–‡ä»¶
          - ğŸ“credible_set.v2.1.20240623: fine-mapping ç»“æœè§£å‹åçš„æ–‡ä»¶å¤¹
            - ğŸ“v2.1: fine-mapping ç»“æœè§£å‹åçš„æ–‡ä»¶å¤¹
              - ğŸ“„credible_set.txt: æ¯ä¸ªæ€§çŠ¶æˆ–ç–¾ç—…çš„è¯¦ç»†å†…å®¹ä¿¡æ¯
              - ğŸ“„meta.txt: æ¯ä¸ªæ€§çŠ¶æˆ–ç–¾ç—…çš„æ¦‚è¿°æ–‡ä»¶
          - ğŸ“„credible_set.v2.1.20240623.tar.gz: ä¸‹è½½ fine-mapping ç»“æœçš„å‹ç¼©æ–‡ä»¶
      - ğŸ“FinnGen: æ‰€æœ‰ FinnGen é˜Ÿåˆ— fine-mapping ç»“æœæ–‡ä»¶çš„æ¥æºæ•°æ®
        - ğŸ“decompression: å¯¹æ‰€æœ‰ä¸‹è½½çš„æ–‡ä»¶è¿›è¡Œå‹ç¼©
          - ğŸ“„`finngen_R11_{trait}.txt`: è§£å‹åçš„æ–‡ä»¶
        - ğŸ“download: ä¸‹è½½ FinnGen çš„æ‰€æœ‰ fine-mapping ç»“æœæ–‡ä»¶
          - ğŸ“„`finngen_R11_{trait}.gz`: ä¸‹è½½çš„å‹ç¼©æ–‡ä»¶
      - ğŸ“UKBB: æ‰€æœ‰ UKBB é˜Ÿåˆ— fine-mapping ç»“æœæ–‡ä»¶çš„æ¥æºæ•°æ®
        - ğŸ“decompression: å¯¹ UKBB_94traits_release1.bed.gz æ–‡ä»¶è¿›è¡Œå‹ç¼©
          - ğŸ“„UKBB_94traits_release1.bed: è§£å‹åçš„æ–‡ä»¶
        - ğŸ“download: ä¸‹è½½ UKBB çš„ fine-mapping ç»“æœæ–‡ä»¶
          - ğŸ“release1.1: å¯¹ä¸‹è½½ fine-mapping ç»“æœæ–‡ä»¶è¿›è¡Œè§£å‹
            - ğŸ“„README.html
            - ğŸ“„README.md
            - ğŸ“„UKBB_94traits_release1.bed.gz
            - ğŸ“„UKBB_94traits_release1.bed.gz.tbi
            - ğŸ“„UKBB_94traits_release1.cols
            - ğŸ“„UKBB_94traits_release1.traits
            - ğŸ“„UKBB_94traits_release1_regions.bed.gz
            - ğŸ“„UKBB_94traits_release1_regions.bed.gz.tbi
            - ğŸ“„UKBB_94traits_release1_regions.cols
          - ğŸ“„UKBB_94traits_release1.1.tar.gz: ä¸‹è½½ fine-mapping ç»“æœæ–‡ä»¶
    - ğŸ“output: å¯¹æ‰€æœ‰ä¸‹è½½åè§£å‹çš„æ–‡ä»¶è¿›è¡Œå¤„ç†æˆç»Ÿä¸€çš„è§„èŒƒæ ¼å¼å­˜å‚¨è·¯å¾„
      - ğŸ“`{source}`: ä¸åŒçš„æ¥æºé˜Ÿåˆ—
        - ğŸ“finemap: å½¢æˆ SCAVENGE çš„ç®—æ³•è¾“å…¥æ ¼å¼, ä»¥åŠè·‘ HOMER, MAGMA çš„è¾“å…¥æ ¼å¼
        - ğŸ“trait: å½¢æˆå¯¹æ€§çŠ¶æˆ–è€…ç–¾ç—…è¯¦ç»†å†…å®¹çš„æ–‡ä»¶
    - ğŸ“filter: è¿™é‡Œå­˜å‚¨çš„æ˜¯é€šè¿‡ä¸€å®šè§„åˆ™è¿‡æ»¤ fine-mapping æ•°æ®å¹¶ä¸”é€šè¿‡ liftOver è¿›è¡Œè½¬åŒ–ä¸ºå¯¹åº”çš„å‚è€ƒåŸºå› ç»„æ•°æ®
      - ğŸ“finemap: output æ–‡ä»¶å¤¹å†…çš„ finemap è¿‡æ»¤åçš„æ•°æ®
      - ğŸ“input_hg19: éœ€è¦è½¬åŒ–ä¸º hg38 çš„è¾“å…¥æ€§çŠ¶æˆ–ç–¾ç—…æ•°æ®
      - ğŸ“input_hg38: éœ€è¦è½¬åŒ–ä¸º hg19 çš„è¾“å…¥æ€§çŠ¶æˆ–ç–¾ç—…æ•°æ®
      - ğŸ“output_hg19: å·²ç»æ˜¯ hg19 çš„å’Œè½¬åŒ–ä¸º hg19 çš„è¾“å‡ºæ•°æ®
      - ğŸ“output_hg38: å·²ç»æ˜¯ hg38 çš„å’Œè½¬åŒ–ä¸º hg38 çš„è¾“å‡ºæ•°æ®
      - ğŸ“result_hg19: hg38 çš„è¾“å‡ºæ•°æ®è½¬åŒ–ä¸ºç»Ÿä¸€çš„è§„èŒƒæ ¼å¼æ•°æ®
      - ğŸ“result_hg38: hg19 çš„è¾“å‡ºæ•°æ®è½¬åŒ–ä¸ºç»Ÿä¸€çš„è§„èŒƒæ ¼å¼æ•°æ®
      - ğŸ“trait: output æ–‡ä»¶å¤¹å†…çš„ trait è¿‡æ»¤åçš„æ•°æ®
      - ğŸ“unmap_hg19: è½¬åŒ–ä¸º hg19 çš„æœªæ˜ å°„çš„æ•°æ®
      - ğŸ“unmap_hg38: è½¬åŒ–ä¸º hg38 çš„æœªæ˜ å°„çš„æ•°æ®
    - ğŸ“finish: fine-mapping ç»“æœæ•°æ®ç»Ÿä¸€å¤„ç†åçš„æœ€ç»ˆæ–‡ä»¶, æ­¤æ•°æ®ç”¨äºè·‘ g-chromVAR, SCAVENGE, HOMER, MAGMA ç­‰æ–¹æ³•çš„è¾“å…¥æ–‡ä»¶
      - ğŸ“hg19: å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„ fine-mapping ç»“æœæœ€ç»ˆæ•°æ®
        - ğŸ“„`{trait_label}.bed`: æ€§çŠ¶æˆ–ç–¾ç—…ç®—æ³•çš„è¾“å…¥æ•°æ®
      - ğŸ“hg38: å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„ fine-mapping ç»“æœæœ€ç»ˆæ•°æ®
        - ğŸ“„`{trait_label}.bed`: æ€§çŠ¶æˆ–ç–¾ç—…ç®—æ³•çš„è¾“å…¥æ•°æ®
      - ğŸ“trait: fine-mapping ç»“æœæ•°æ®ç»Ÿä¸€å¤„ç†åçš„æœ€ç»ˆæ–‡ä»¶, æ­¤è·¯å¾„ä¸‹çš„æ˜¯è¯¦ç»†çš„æ€§çŠ¶æˆ–ç–¾ç—…çš„å†…å®¹ä¿¡æ¯
        - ğŸ“„`{trait_label}.txt`: æ€§çŠ¶æˆ–ç–¾ç—…çš„è¯¦ç»†å†…å®¹ä¿¡æ¯
      - ğŸ“„variant_id.txt: å­˜å‚¨çªå˜ä½ç‚¹å’Œ rsId æ˜ å°„å…³ç³»æ–‡ä»¶
      - ğŸ“„variant_id_tmp.txt: å­˜å‚¨ä¸´æ—¶çªå˜ä½ç‚¹å’Œ rsId æ˜ å°„å…³ç³»æ–‡ä»¶
    - ğŸ“homer: æ‰€æœ‰æ€§çŠ¶æˆ–ç–¾ç—…è·‘ HOMER çš„ç»“æœæ•°æ®
      - ğŸ“hg19: å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„æ‰€æœ‰æ€§çŠ¶æˆ–ç–¾ç—…è·‘ HOMER çš„ç»“æœæ•°æ®
        - ğŸ“`{trait_label}`: å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„æ­¤æ€§çŠ¶æˆ–ç–¾ç—…è·‘ HOMER çš„ç»“æœæ•°æ®
      - ğŸ“hg38: å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„æ‰€æœ‰æ€§çŠ¶æˆ–ç–¾ç—…è·‘ HOMER çš„ç»“æœæ•°æ®
        - ğŸ“`{trait_label}`: å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„æ­¤æ€§çŠ¶æˆ–ç–¾ç—…è·‘ HOMER çš„ç»“æœæ•°æ®
    - ğŸ“magma: æ‰€æœ‰æ€§çŠ¶æˆ–ç–¾ç—…è·‘ MAGMA çš„ç»“æœæ•°æ®
      - ğŸ“gene: MAGMA å‚è€ƒåŸºå› æ•°æ®
        - ğŸ“„gene_hg19.bed
        - ğŸ“„gene_hg38.bed
      - ğŸ“magma_input: å½¢æˆçš„ MAGMA è¾“å…¥çš„æ–‡ä»¶æ•°æ®ä¿¡æ¯
        - ğŸ“g1000_afr: African å‚è€ƒèƒŒæ™¯æ¨¡æ¿æ•°æ®
        - ğŸ“g1000_amr: American å‚è€ƒèƒŒæ™¯æ¨¡æ¿æ•°æ®
        - ğŸ“g1000_eas: East Asian å‚è€ƒèƒŒæ™¯æ¨¡æ¿æ•°æ®
        - ğŸ“g1000_eur: European å‚è€ƒèƒŒæ™¯æ¨¡æ¿æ•°æ®
        - ğŸ“g1000_sas: South Asian å‚è€ƒèƒŒæ™¯æ¨¡æ¿æ•°æ®
        - ğŸ“hg19: å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„ MAGMA è¾“å…¥çš„æ–‡ä»¶
          - `{trait_label}.bim`: ç”¨äºç¬¬ä¸€æ­¥çš„åŸºå› åŒ¹é… SNP ä½ç‚¹
          - `{trait_label}.txt`: ç”¨äºç¬¬äºŒæ­¥çš„åŸºå› å¯Œé›†æ³¨é‡Š
        - ğŸ“hg38: å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„ MAGMA è¾“å…¥çš„æ–‡ä»¶
          - `{trait_label}.bim`: ç”¨äºç¬¬ä¸€æ­¥çš„åŸºå› åŒ¹é… SNP ä½ç‚¹
          - `{trait_label}.txt`: ç”¨äºç¬¬äºŒæ­¥çš„åŸºå› å¯Œé›†æ³¨é‡Š
      - ğŸ“magma_output: 
        - ğŸ“hg19_anno: å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„ MAGMA ç¬¬ä¸€æ­¥çš„è¾“å‡ºæ–‡ä»¶
        - ğŸ“hg19_gene: å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„ MAGMA ç¬¬äºŒæ­¥çš„è¾“å‡ºæ–‡ä»¶
        - ğŸ“hg38_anno: å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„ MAGMA ç¬¬ä¸€æ­¥çš„è¾“å‡ºæ–‡ä»¶
        - ğŸ“hg38_gene: å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„ MAGMA ç¬¬äºŒæ­¥çš„è¾“å‡ºæ–‡ä»¶

```shell
source_path="/public/home/lcq/rgzn/yuzhengmin/keti"
target_path="/mnt/data3/workspace/SCVdb"

# scATAC-seq
scp -r "$source_path/database/sc_variant/table/download/scatac" "root@bio.liclab.net:$target_path/data/data/download/"

# HOMER
scp -r "$source_path/database/sc_variant/table/homer/hg19" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/homer/"
scp -r "$source_path/database/sc_variant/table/homer/hg38" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/homer/"
scp -r "$source_path/database/sc_variant/table/homer/t_homer_trait_tf_hg19.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/homer/"
scp -r "$source_path/database/sc_variant/table/homer/t_homer_trait_tf_hg38.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/homer/"
scp -r "$source_path/database/sc_variant/table/homer/t_homer_tf_trait_count.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/homer/"
scp -r "$source_path/database/sc_variant/table/homer/trait_tf" "root@bio.liclab.net:$target_path/mysql/mysqlfile/"

# MAGMA
scp -r "$source_path/database/sc_variant/table/magma/hg19" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/magma/"
scp -r "$source_path/database/sc_variant/table/magma/hg19_anno" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/magma/"
scp -r "$source_path/database/sc_variant/table/magma/hg38" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/magma/"
scp -r "$source_path/database/sc_variant/table/magma/hg38_anno" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/magma/"
scp -r "$source_path/database/sc_variant/table/magma/gene_enrichment_trait_table" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/magma/"
scp -r "$source_path/database/sc_variant/table/magma/t_magma_hg19.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/magma/"
scp -r "$source_path/database/sc_variant/table/magma/t_magma_hg38.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/magma/"
scp -r "$source_path/database/sc_variant/table/magma/t_magma_gene_trait_count.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/magma/"
scp -r "$source_path/database/sc_variant/table/magma/trait_gene" "root@bio.liclab.net:$target_path/mysql/mysqlfile/"

scp -r "$source_path/database/sc_variant/table/trait_variant_overlap/sample" "root@bio.liclab.net:$target_path/mysql/mysqlfile/trait_sample/"
scp -r "$source_path/database/sc_variant/table/trait_variant_overlap/trait" "root@bio.liclab.net:$target_path/mysql/mysqlfile/trait_sample"

# scATAC-seq
scp -r "$source_path/database/sc_variant/table/scatac/difference_gene" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/"
scp -r "$source_path/database/sc_variant/table/scatac/difference_tf" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/"
scp -r "$source_path/database/sc_variant/table/scatac/gene_enrichment_table" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/"
scp -r "$source_path/database/sc_variant/table/scatac/difference_gene_data.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/"
scp -r "$source_path/database/sc_variant/table/scatac/difference_tf_data.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/"
scp -r "$source_path/database/sc_variant/table/scatac/sample_gene_data.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/"

# Overlap
scp -r "$source_path/database/sc_variant/table/trait_variant_overlap/gchromvar_sample_enrichment.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/trait_sample/"
scp -r "$source_path/database/sc_variant/table/trait_variant_overlap/scavenge_sample_enrichment.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/trait_sample/"
scp -r "$source_path/database/sc_variant/table/trait_variant_overlap/trs_overlap_gchromvar.h5ad" "root@bio.liclab.net:$target_path/data/data/trait_variant_overlap/"
scp -r "$source_path/database/sc_variant/table/trait_variant_overlap/trs_overlap_scavenge.h5ad" "root@bio.liclab.net:$target_path/data/data/trait_variant_overlap/"

# variant
scp -r "$source_path/database/sc_variant/table/variant" "root@bio.liclab.net:$target_path/mysql/mysqlfile/"

# TRS
scp -r "$source_path/database/sc_variant/table/trs" "root@bio.liclab.net:$target_path/data/data/"
scp -r "$source_path/database/sc_variant/table/trs_big" "root@bio.liclab.net:$target_path/data/data/download/"

# HOMER
scp -r "$source_path/variant/homer" "root@bio.liclab.net:$target_path/data/data/"
scp -r "$source_path/variant/homer.tar.gz" "root@bio.liclab.net:$target_path/data/data/download/magma_homer/"

# MAGMA
scp -r $source_path/variant/magma/magma_output/hg19_anno/*.genes.annot root@bio.liclab.net:"$target_path/data/data/magma/magma_output/hg19_anno/"
scp -r $source_path/variant/magma/magma_output/hg19_gene/*.genes.out root@bio.liclab.net:"$target_path/data/data/magma/magma_output/hg19_gene/"
scp -r $source_path/variant/magma/magma_output/hg38_anno/*.genes.annot root@bio.liclab.net:"$target_path/data/data/magma/magma_output/hg38_anno/"
scp -r $source_path/variant/magma/magma_output/hg38_gene/*.genes.out root@bio.liclab.net:"$target_path/data/data/magma/magma_output/hg38_gene/"
scp -r "$source_path/variant/magma/magma_output.tar.gz" "root@bio.liclab.net:$target_path/data/data/download/magma_homer/"
scp -r "$source_path/variant/magma/magma_output.tar.gz" "root@bio.liclab.net:$target_path/data/data/download/magma_homer/"

# Difference Gene/TF
scp -r "$source_path/database/sc_variant/table/scatac/difference_gene_data.txt" "root@bio.liclab.net:$target_path/data/data/download/difference/"
scp -r "$source_path/database/sc_variant/table/scatac/difference_tf_data.txt" "root@bio.liclab.net:$target_path/data/data/download/difference/"
scp -r "$source_path/database/sc_variant/table/scatac/difference_gene" "root@bio.liclab.net:$target_path/data/data/download/difference/"
scp -r "$source_path/database/sc_variant/table/scatac/difference_gene" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/"
scp -r "$source_path/database/sc_variant/table/scatac/difference_tf" "root@bio.liclab.net:$target_path/data/data/download/difference/"
scp -r "$source_path/database/sc_variant/table/scatac/difference_tf" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/"
scp -r "$source_path/database/sc_variant/table/scatac/sample_gene" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/"
scp -r "$source_path/database/sc_variant/table/scatac/sample_tf" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/"
scp -r $source_path/scATAC/**/**/**/*_difference_gene.h5ad root@bio.liclab.net:"$target_path/data/data/download/difference/difference_gene_h5ad/"
scp -r $source_path/scATAC/**/**/**/*_tf_activity_data.h5ad root@bio.liclab.net:"$target_path/data/data/download/difference/difference_tf_h5ad/"

# Difference gene enrichment
scp -r "$source_path/database/sc_variant/table/scatac/gene_enrichment_table_data.tar.gz" "root@bio.liclab.net:$target_path/data/data/download/enrichment/"
scp -r "$source_path/database/sc_variant/table/scatac/gene_enrichment_table" "root@bio.liclab.net:$target_path/data/data/download/enrichment/"
scp -r $source_path/scATAC/**/**/**/*_gene_enrichment.txt root@bio.liclab.net:"$target_path/data/data/download/enrichment/scatac/complete/"

# variant
scp -r "$source_path/variant/finish/fine_mapping_hg19.tar.gz" "root@bio.liclab.net:$target_path/data/data/download/trait/"
scp -r "$source_path/variant/finish/fine_mapping_hg38.tar.gz" "root@bio.liclab.net:$target_path/data/data/download/trait/"
scp -r "$source_path/variant/finish/fine_mapping_trait.tar.gz" "root@bio.liclab.net:$target_path/data/data/download/trait/"

# Trait gene enrichment
scp -r "$source_path/database/sc_variant/table/magma/gene_enrichment_trait/gene_enrichment_trait_hg19.tar.gz" "root@bio.liclab.net:$target_path/data/data/download/enrichment/"
scp -r "$source_path/database/sc_variant/table/magma/gene_enrichment_trait/gene_enrichment_trait_hg38.tar.gz" "root@bio.liclab.net:$target_path/data/data/download/enrichment/"

scp -r "$source_path/database/sc_variant/table/magma/gene_enrichment_trait/hg19" "root@bio.liclab.net:$target_path/data/data/download/enrichment/trait/"
scp -r "$source_path/database/sc_variant/table/magma/gene_enrichment_trait/hg38" "root@bio.liclab.net:$target_path/data/data/download/enrichment/trait/"

# GENE
scp -r $source_path/gene/annotation/dbSNP/common_snp_chunk/* root@bio.liclab.net:"$target_path/mysql/mysqlfile/gene/annotation/dbSNP/common_snp_chunk/"
scp -r $source_path/gene/annotation/dbSUPER/*.txt root@bio.liclab.net:"$target_path/mysql/mysqlfile/gene/annotation/dbSUPER/"
scp -r $source_path/gene/annotation/GTEx/eqtl_chunk/* root@bio.liclab.net:"$target_path/mysql/mysqlfile/gene/annotation/GTEx/eqtl_chunk/"
scp -r $source_path/gene/annotation/gwasATLAS/gwasatlas_v20191115_risk_snp_*.txt root@bio.liclab.net:"$target_path/mysql/mysqlfile/gene/annotation/gwasATLAS/"
scp -r $source_path/gene/annotation/SEA/sea_v3_*.txt root@bio.liclab.net:"$target_path/mysql/mysqlfile/gene/annotation/SEA/"
scp -r $source_path/gene/annotation/SEdb/sedb_v2_super_enhancer_hg19.txt root@bio.liclab.net:"$target_path/mysql/mysqlfile/gene/annotation/SEdb/"
scp -r $source_path/gene/annotation/SEdb/sedb_v2_super_enhancer_hg38.txt root@bio.liclab.net:"$target_path/mysql/mysqlfile/gene/annotation/SEdb/"
scp -r $source_path/gene/annotation/SEdb/enhancer_chunk/* root@bio.liclab.net:"$target_path/mysql/mysqlfile/gene/annotation/SEdb/enhancer_chunk/"

scp -r $source_path/gene/annotation/dbSNP/dbsnp_common_snp_hg19.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg19/"
scp -r $source_path/gene/annotation/dbSNP/dbsnp_common_snp_hg38.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg38/"
scp -r $source_path/gene/annotation/dbSUPER/dbsuper_super_enhancer_hg19.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg19/"
scp -r $source_path/gene/annotation/dbSUPER/dbsuper_super_enhancer_hg38.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg38/"
scp -r $source_path/gene/annotation/GTEx/gtex_v10_eqtl_hg19.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg19/"
scp -r $source_path/gene/annotation/GTEx/gtex_v10_eqtl_hg38.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg38/"
scp -r $source_path/gene/annotation/gwasATLAS/gwasatlas_v20191115_risk_snp_hg19.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg19/"
scp -r $source_path/gene/annotation/gwasATLAS/gwasatlas_v20191115_risk_snp_hg38.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg38/"
scp -r $source_path/gene/annotation/SEA/sea_v3_enhancer_hg19.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg19/"
scp -r $source_path/gene/annotation/SEA/sea_v3_enhancer_hg38.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg38/"
scp -r $source_path/gene/annotation/SEA/sea_v3_super_enhancer_hg19.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg19/"
scp -r $source_path/gene/annotation/SEA/sea_v3_super_enhancer_hg38.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg38/"
scp -r $source_path/gene/annotation/SEdb/sedb_v2_enhancer_hg19.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg19/"
scp -r $source_path/gene/annotation/SEdb/sedb_v2_enhancer_hg38.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg38/"
scp -r $source_path/gene/annotation/SEdb/sedb_v2_super_enhancer_hg19.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg19/"
scp -r $source_path/gene/annotation/SEdb/sedb_v2_super_enhancer_hg38.tar.gz root@bio.liclab.net:"$target_path/data/data/download/gene/hg38/"

```
