# SCVDB å¤„ç†è¿‡ç¨‹

> å¯¹æ–‡ä»¶å¤¹çš„å™è¿°

| æ–‡ä»¶å¤¹          | è¯´æ˜                        |
|--------------|---------------------------|
| database     | å½¢æˆæ•°æ®åº“æ–‡ä»¶                   |
| find_mapping | find_mapping æµç¨‹           |
| fragments    | å°†ä¸‰ä¸ªæ–‡ä»¶å¤„ç†æˆ fragments æ–‡ä»¶     |
| variant      | å¯¹ä¸‹è½½çš„çªå˜æ•°æ®è¿›è¡Œå¤„ç†              |
| liftOver     | å°† hg19 çªå˜æ•°æ®è½¬åŒ–ä¸º hg38 çªå˜æ•°æ®  |
| R            | å°†çªå˜æ•°æ®å’Œ scATAC æ•°æ®æ•´åˆåç»“æœå¤„ç†å†…å®¹ |
| uti          | å…¬å…±æ–¹æ³•ç±»                     |

## 1. æ”¶é›† scATAC å’Œ variant æ•°æ®

### 1.1 æ”¶é›† scATAC æ•°æ®

> åœ¨ **scATACRef** æ”¶é›†

### 1.2 æ”¶é›† **variant** æ•°æ®

> æ”¶é›†çš„æ•°æ®ä¿¡æ¯

| èµ„æº                                | æ¥æº                             | PMID     | æ€»å…±æ€§çŠ¶æ•°é‡ | ç­›é€‰åæ€§çŠ¶æ•°é‡ | æ•°æ®é“¾æ¥                                                                | genome |
|-----------------------------------|--------------------------------|----------|--------|---------|---------------------------------------------------------------------|--------|
| GWAS catalog fine-mapping results | Annalisa Buniello et al., 2019 | 30445434 | 381    | 3       | https://www.ebi.ac.uk/gwas/downloads/summary-statistics             | hg38   |
| UKBB fine-mapping results         | Kanai, M. et al., 2021         |          | 94     | 94      | https://www.finucanelab.org/data                                    | hg19   |
| FinnGen fine-mapping results      | Kanai, M. et al., 2022         | 36653562 | 86     | 51      | https://www.finngen.fi/en/access_results                            | hg38   |
| BBJ fine-mapping results          | Kanai, M. et al., 2021         |          | 79     | 77      | https://humandbs.biosciencedbc.jp/en/hum0197-v15#hum0197.v5.gwas.v1 | hg19   |
| CAUSALdb fine-mapping results     | Jianhua Wang et al., 2019      | 31691819 | 3292   | 2144    | http://www.mulinlab.org/causaldb/index.html                         | hg19   |

> çªå˜æ•°æ®è½¬åŒ–ä¸ºæŒ‡å®šæ ¼å¼ä¿¡æ¯
>
> åœ¨è½¬åŒ–è¿‡ç¨‹ä¸­é‡åˆ°å¤šç§æ–¹æ³•è·å– PIP ä¸ªå€¼çš„æ—¶å€™ï¼Œåªé€‰å– fine-mapping å¾—åˆ°çš„ PIP å€¼

> æ¯åˆ—çš„å«ä¹‰:  `chr	postion	postion	variant_name	PIP trait`
>
> ä¾‹å¦‚:

```bed
chr11	305619	305619	rs6421984	0.106024	baso
chr11	306920	306920	rs7115703	0.108784	baso
chr11	307244	307244	rs10751648	0.211543	baso
chr11	307696	307696	rs7480524	0.131486	baso
```

> ä¹‹åçªå˜æ•°æ®é€šè¿‡ liftOver è¿›è¡Œè½¬åŒ–ä¸º hg38

## 2. å¯¹çªå˜æ•°æ®å½¢æˆæ•°æ®åº“è¡¨æ ¼å¤„ç†

> çªå˜æ•°æ®çš„æ–‡ä»¶åæ˜¯å­˜å‚¨æ•°æ®è¡¨æ ¼ä¸­çš„ id åŠ ä¸Šå‚è€ƒåŸºå› ç»„

## 3. å¯¹ Fragments æ–‡ä»¶è¿›è¡Œ ArchR å¤„ç†

> å¾—åˆ° scATAC æ•°æ®è¡¨è¾¾æ•°æ®

## 4. å°† scATAC å’Œçªå˜æ•°æ®æ•´åˆ

> ä½¿ç”¨ SCAVENGE æ–¹æ³•è¿›è¡Œæ•´åˆ

## 5. è·¯å¾„å­˜å‚¨è¯´æ˜

> `path` ä¸ºæ ¹è·¯å¾„

- ğŸ“`{path}`
  - ğŸ“database: å­˜å‚¨ `SCVdb` æ•°æ®åº“æ•°æ®
    - ğŸ“code: å­˜å‚¨è·‘æ•°æ®åº“æ•°æ®çš„ä»£ç 
      - ğŸ“`{server}`: ä»£è¡¨ä¸åŒçš„æœåŠ¡å™¨, å­˜å‚¨æ‰§è¡Œ g-chromVAR å’Œ SCAVENGE æ–¹æ³•çš„ä»£ç 
        - ğŸ“„library.R: å¼•ç”¨çš„ R åŒ…
        - ğŸ“„static_function.R: å•ç»†èƒå’Œæ•´åˆä»£ç çš„å¤„ç†
        - ğŸ“„integration.R: å°†å¤šä¸ªæ€§çŠ¶æˆ–ç–¾ç—…å¾ªç¯è·‘æ•°æ®
        - ğŸ“„run.R: æ‰§è¡Œçš„æ€»ä»£ç 
    - ğŸ“sc_variant: å­˜å‚¨æ®åº“æ•°æ®è¾“å‡ºçš„æ ¹è·¯å¾„
      - ğŸ“result: å­˜å‚¨ g-chromVAR å’Œ SCAVENGE æ–¹æ³•ç»“æœ
        - ğŸ“`{scATAC-seq}`: ä»£è¡¨ä¸åŒçš„å•ç»†èƒæ•°æ®, å­˜å‚¨æŸä¸ªå•ç»†èƒæ•°æ®æ•´åˆæ‰€æœ‰çš„æ€§çŠ¶æˆ–ç–¾ç—…çš„ç»“æœæ•°æ®
          - ğŸ“„`{scATAC-seq}__{genome}__{trait_label_file}__mat.txt`: å­˜å‚¨ TRS, å¯Œé›†çŠ¶æ€ç­‰ä¿¡æ¯æ–‡ä»¶
          - ğŸ“„`{scATAC-seq}__{genome}__{trait_label_file}__mat_info.rda`: å­˜å‚¨äº†ç»“æœ txt å’Œ mutual-KNN ä¿¡æ¯æ–‡ä»¶
      - ğŸ“scATAC: å­˜å‚¨äº†å•ç»†èƒè·‘ä¹‹å‰çš„å¤„ç†æ–‡ä»¶ä»¥åŠè·‘ä¹‹åç»“æœæ–‡ä»¶
        - ğŸ“`{scATAC-seq}`: ä»£è¡¨ä¸åŒçš„å•ç»†èƒæ•°æ®, å­˜å‚¨æŸä¸ªå•ç»†èƒæ•°æ®è·‘ä¹‹å‰çš„å¤„ç†æ–‡ä»¶ä»¥åŠè·‘ä¹‹åç»“æœæ–‡ä»¶
          - ğŸ“„`{scATAC-seq}.txt`: ç»†èƒæ³¨é‡Šæ–‡ä»¶
          - ğŸ“„`{scATAC-seq}_all.rda`: å•ç»†èƒå¤„ç†æ–‡ä»¶, åœ¨ä¸æ€§çŠ¶æˆ–è€…ç–¾ç—…è¿›è¡Œæ•´åˆè¯»å–çš„æ˜¯è¿™ä¸ªæ–‡ä»¶
          - ğŸ“„`{scATAC-seq}_SE_gvar_SE_gvar_bg.rda`: å•ç»†èƒå¤„ç†çš„ä¸­é—´æ–‡ä»¶, å†…å®¹æ¯” ${scATAC-seq}_all.rda è¿™ä¸ªæ–‡ä»¶å°‘
          - ğŸ“„`{scATAC-seq}_trs_scavenge_data.h5ad`: æŸä¸ªå•ç»†èƒæ•°æ®æ•´åˆæ‰€æœ‰çš„æ€§çŠ¶æˆ–ç–¾ç—…çš„ç»“æœæ•°æ®å½¢æˆä¸€ä¸ª h5ad æ–‡ä»¶æ ¼å¼
      - ğŸ“table: å­˜å‚¨ä¸æ•°æ®åº“æ­å»ºç›¸å…³çš„æ•°æ®
        - ğŸ“download: å­˜å‚¨é€šè¿‡ SnapATAC2 å¤„ç†åçš„ scATAC-seq æ•°æ®
          - ğŸ“scatac: å­˜å‚¨é€šè¿‡ SnapATAC2 å¤„ç†åçš„ scATAC-seq æ•°æ®
            - ğŸ“„`{scATAC-seq}_snapATAC2.h5ad`: åœ¨ç½‘ç«™ä¸Šä¸‹è½½é¡µä¸‹è½½çš„æ•°æ®
          - ğŸ“„cp.sh: å°†é€šè¿‡ SnapATAC2 å¤„ç†åçš„ scATAC-seq æ•°æ®éƒ½å¤åˆ¶åˆ°é€šè·¯å¾„ä¸‹çš„ scatac ä¸­
        - ğŸ“homer: å­˜å‚¨ HOMER è·‘å‡ºæ¥çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„è½¬å½•å› å­æ•°æ®
          - ğŸ“hg19: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„è½¬å½•å› å­æ•°æ®
            - ğŸ“„`t_homer_{group}`.txt:  ä¸æ•°æ®åº“ä¸­ `t_homer_hg19_{group}` è¡¨å¯¹åº”
          - ğŸ“hg38: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„è½¬å½•å› å­æ•°æ®
            - ğŸ“„`t_homer_{group}`.txt:  ä¸æ•°æ®åº“ä¸­ `t_homer_hg38_{group}` è¡¨å¯¹åº”
          - ğŸ“„t_homer_trait_tf.txt: å­˜å‚¨åœ¨ä¸åŒå‚è€ƒåŸºå› ç»„ä¸‹æ€§çŠ¶æˆ–ç–¾ç—…æ ·æœ¬ä¸è½¬å½•å› å­çš„æ˜ å°„å…³ç³», ä¸æ•°æ®åº“ä¸­ `t_trait_tf` è¡¨å¯¹åº”
          - ğŸ“„trait_tf_hg19.txt: è¿™ä¸ªæ–‡ä»¶æ˜¯å¯¹ `hg19` æ–‡ä»¶æ•°æ®æ”¾åœ¨äº†ä¸€èµ· (æ²¡æœ‰ç”¨åˆ°, ä¿ç•™)
          - ğŸ“„trait_tf_hg38.txt: è¿™ä¸ªæ–‡ä»¶æ˜¯å¯¹ `hg38` æ–‡ä»¶æ•°æ®æ”¾åœ¨äº†ä¸€èµ· (æ²¡æœ‰ç”¨åˆ°, ä¿ç•™)
        - ğŸ“magma: å­˜å‚¨ MAGMA è·‘å‡ºæ¥çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› æ•°æ®
          - ğŸ“gene_enrichment_trait: å­˜å‚¨æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› çš„ GO, KEGG ç­‰åŸºå› å¯Œé›†åŠŸèƒ½
            - ğŸ“hg19: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› çš„ GO, KEGG ç­‰åŸºå› å¯Œé›†åŠŸèƒ½
            - ğŸ“hg38: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› çš„ GO, KEGG ç­‰åŸºå› å¯Œé›†åŠŸèƒ½
          - ğŸ“gene_enrichment_trait_table: å­˜å‚¨æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› çš„ GO, KEGG ç­‰åŸºå› å¯Œé›†åŠŸèƒ½
            - ğŸ“hg19: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå›  GO, KEGG ç­‰åŸºå› å¯Œé›†åŠŸèƒ½
              - ğŸ“„`t_gene_enrichment_trait_hg19_{group}.txt`: ä¸æ•°æ®åº“ä¸­ `t_gene_enrichment_trait_hg19_{group}` è¡¨å¯¹åº”
            - ğŸ“hg38: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå›  GO, KEGG ç­‰åŸºå› å¯Œé›†åŠŸèƒ½
              - ğŸ“„`t_gene_enrichment_trait_hg38_{group}.txt`: ä¸æ•°æ®åº“ä¸­ `t_gene_enrichment_trait_hg38_{group}` è¡¨å¯¹åº”
          - ğŸ“hg19: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› æ•°æ® (å¸¦æœ‰å¯Œé›†çš„å¾—åˆ†)
            - ğŸ“„`t_magma_{group}.txt`: ä¸æ•°æ®åº“ä¸­ `t_magma_hg19_{group}` è¡¨å¯¹åº”
          - ğŸ“hg19_anno: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› æ•°æ® (å¸¦æœ‰çªå˜å’ŒåŸºå› æ˜ å°„)
            - ğŸ“„`t_magma_{group}.txt`: ä¸æ•°æ®åº“ä¸­ `t_magma_anno_hg19_{group}` è¡¨å¯¹åº”
          - ğŸ“hg38: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› æ•°æ® (å¸¦æœ‰å¯Œé›†çš„å¾—åˆ†)
            - ğŸ“„`t_magma_{group}.txt`: ä¸æ•°æ®åº“ä¸­ `t_magma_hg38_{group}` è¡¨å¯¹åº”
          - ğŸ“hg38_anno: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„æ€§çŠ¶æˆ–ç–¾ç—…æ„Ÿå…´è¶£çš„åŸºå› æ•°æ® (å¸¦æœ‰çªå˜å’ŒåŸºå› æ˜ å°„)
            - ğŸ“„`t_magma_{group}.txt`: ä¸æ•°æ®åº“ä¸­ `t_magma_anno_hg38_{group}` è¡¨å¯¹åº”
          - ğŸ“„t_magma.txt: å­˜å‚¨åœ¨ä¸åŒå‚è€ƒåŸºå› ç»„ä¸‹æ€§çŠ¶æˆ–ç–¾ç—…æ ·æœ¬ä¸åŸºå› çš„æ˜ å°„å…³ç³», ä¸æ•°æ®åº“ä¸­ `t_trait_gene` è¡¨å¯¹åº”
        - ğŸ“scatac: å­˜å‚¨ä¸ scATAC-seq æ•°æ®ç›¸å…³çš„æ•°æ®åº“å†…å®¹
          - ğŸ“difference_gene: å•ç»†èƒæ ·æœ¬çš„å·®å¼‚åŸºå› å¯Œé›†
            - `{scATAC-seq}_difference_gene_data.txt`: æŸä¸ªå•ç»†èƒæ ·æœ¬çš„å·®å¼‚åŸºå› å¯Œé›† (æ²¡æœ‰ç”¨åˆ°, ä¿ç•™)
          - ğŸ“difference_tf: å•ç»†èƒæ ·æœ¬çš„å·®å¼‚è½¬å½•å› å­å¯Œé›†
            - `{scATAC-seq}_difference_tf_data.txt`: æŸä¸ªå•ç»†èƒæ ·æœ¬çš„å·®å¼‚è½¬å½•å› å­å¯Œé›† (æ²¡æœ‰ç”¨åˆ°, ä¿ç•™)
          - ğŸ“gene_enrichment_table: å•ç»†èƒæ ·æœ¬çš„å·®å¼‚åŸºå›  GO, KEGG ç­‰é€šè·¯å¯Œé›†
            - `{scATAC-seq}_gene_enrichment_data.txt`: æŸä¸ªå•ç»†èƒæ ·æœ¬çš„å·®å¼‚åŸºå›  GO, KEGG ç­‰é€šè·¯å¯Œé›†, ä¸æ•°æ®åº“ä¸­ `t_gene_enrichment_{scATAC-seq}` è¡¨å¯¹åº”
          - ğŸ“„difference_gene_data.txt: æ‰€æœ‰å•ç»†èƒæ ·æœ¬å·®å¼‚åŸºå› å¯Œé›†, ä¸æ•°æ®åº“ä¸­ `t_difference_gene` è¡¨å¯¹åº”
          - ğŸ“„difference_tf_data.txt: æ‰€æœ‰å•ç»†èƒæ ·æœ¬å·®å¼‚è½¬å½•å› å­å¯Œé›†, ä¸æ•°æ®åº“ä¸­ `t_difference_tf` è¡¨å¯¹åº”
          - ğŸ“„gene_enrichment_table_data.txt: æ‰€æœ‰å•ç»†èƒæ ·æœ¬çš„å·®å¼‚åŸºå›  GO, KEGG ç­‰é€šè·¯å¯Œé›† (æ²¡æœ‰ç”¨åˆ°, ä¿ç•™)
          - ğŸ“„sample_gene_data.txt: å­˜å‚¨å•ç»†èƒæ ·æœ¬ä¸åŸºå› çš„æ˜ å°„å…³ç³»æ–‡ä»¶, ä¸æ•°æ®åº“ä¸­ `t_sample_gene` è¡¨å¯¹åº”
        - ğŸ“trait_variant_overlap: å­˜å‚¨æ‰€æœ‰å•ç»†èƒå’Œæ‰€æœ‰æ€§çŠ¶æˆ–ç–¾ç—…æ˜¯å¦å­˜åœ¨ overlap çš„æƒ…å†µ (æ˜¯å¦ overlap: !æ˜¯å¦ TRS å¾—åˆ†å…¨ä¸ºé›¶)
          - ğŸ“„gchromvar_sample_enrichment.txt: g-chromVAR æ–¹æ³• overlap æƒ…å†µä»¥ txt æ–‡ä»¶å­˜å‚¨, ä¸æ•°æ®åº“ä¸­ `t_difference_tf` è¡¨å¯¹åº”
          - ğŸ“„scavenge_sample_enrichment.txt: SCAVENGE æ–¹æ³• overlap æƒ…å†µä»¥ txt æ–‡ä»¶å­˜å‚¨, ä¸æ•°æ®åº“ä¸­ `t_difference_tf` è¡¨å¯¹åº”
          - ğŸ“„trs_overlap_gchromvar.h5ad: g-chromVAR æ–¹æ³• overlap æƒ…å†µä»¥ h5ad æ–‡ä»¶å­˜å‚¨ (æ²¡æœ‰ç”¨åˆ°, ä¿ç•™)
          - ğŸ“„trs_overlap_scavenge.h5ad: SCAVENGE æ–¹æ³• overlap æƒ…å†µä»¥ h5ad æ–‡ä»¶å­˜å‚¨ (æ²¡æœ‰ç”¨åˆ°, ä¿ç•™)
        - ğŸ“trs: å­˜å‚¨ç½‘ç«™è¯»å–æ–‡ä»¶å±•ç¤ºå¯è§†åŒ–çš„æ•°æ®å†…å®¹ (è¿™é‡Œå¯¹ç»“æœ h5ad æ–‡ä»¶è¿›è¡Œåˆ†æ–‡ä»¶ 100 ä»½)
          - ğŸ“`{scATAC-seq}`: ä»£è¡¨ä¸åŒçš„å•ç»†èƒæ•°æ®, å­˜å‚¨æŸä¸ªå•ç»†èƒæ•°æ®ä¸åˆ†ç»„æ€§çŠ¶æˆ–è€…ç–¾ç—…çš„ç»“æœ
            - ğŸ“cell_type: ä»¥ç»†èƒç±»å‹ä¸ºåˆ†è¾¨ç‡çš„ TRS ç»“æœ
              - ğŸ“„`{scATAC-seq}_cell_type_trs_gchromvar.h5ad`: g-chromVAR æ–¹æ³•è®¡ç®—å‡ºæ¥çš„ cell types-traits çŸ©é˜µ
              - ğŸ“„`{scATAC-seq}_cell_type_trs_scavenge.h5ad`: SCAVENGE æ–¹æ³•è®¡ç®—å‡ºæ¥çš„ cell types-traits çŸ©é˜µ
            - ğŸ“chunk: TRS ç»“æœ
              - ğŸ“„`{scATAC-seq}_{group}_trs_gchromvar.h5ad`: g-chromVAR æ–¹æ³•è®¡ç®—å‡ºæ¥çš„ cells-traits çŸ©é˜µ
              - ğŸ“„`{scATAC-seq}_{group}_trs_scavenge.h5ad`: SCAVENGE æ–¹æ³•è®¡ç®—å‡ºæ¥çš„ cells-traits çŸ©é˜µ
        - ğŸ“trs_big: å­˜å‚¨ç½‘ç«™ä¸‹è½½çš„æ–‡ä»¶ (è¿™é‡Œå­˜å‚¨ç»“æœ h5ad æ–‡ä»¶, æœªåˆ†è¡¨)
          - ğŸ“`{scATAC-seq}`: ä»£è¡¨ä¸åŒçš„å•ç»†èƒæ•°æ®, å­˜å‚¨æŸä¸ªå•ç»†èƒæ•°æ®ä¸æ‰€æœ‰æ€§çŠ¶æˆ–è€…ç–¾ç—…çš„ç»“æœ
              - ğŸ“„`{scATAC-seq}_trs_gchromvar.h5ad`: g-chromVAR æ–¹æ³•è®¡ç®—å‡ºæ¥çš„ cells-traits çŸ©é˜µ
              - ğŸ“„`{scATAC-seq}_trs_scavenge.h5ad`: SCAVENGE æ–¹æ³•è®¡ç®—å‡ºæ¥çš„ cells-traits çŸ©é˜µ
        - ğŸ“variant: å­˜å‚¨ç½‘ç«™çªå˜ä¿¡æ¯æ•°æ®åº“çš„æ•°æ®
          - ğŸ“hg19: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„ fine-mapping ç»“æœçš„æ•°æ®
            - ğŸ“„`t_variant_{group}_hg19.txt`: ä¸æ•°æ®åº“ä¸­ `t_variant_hg19_{group}` è¡¨å¯¹åº”
          - ğŸ“hg38: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„ fine-mapping ç»“æœçš„æ•°æ®
            - ğŸ“„`t_variant_{group}_hg38.txt`: ä¸æ•°æ®åº“ä¸­ `t_variant_hg38_{group}` è¡¨å¯¹åº”
          - ğŸ“„t_trait_chr_count.txt: å­˜å‚¨æ€§çŠ¶æˆ–è€…ç–¾ç—…ä¸­ä¸åŒæŸ“è‰²è´¨çš„æ•°é‡, ä¸æ•°æ®åº“ä¸­ `t_trait_chr_count` è¡¨å¯¹åº”
      - ğŸ“variant: å­˜å‚¨è·‘æ•°æ®çš„è¾“å…¥ fine-mapping ç»“æœçš„æ•°æ®
        - ğŸ“hg19: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„ fine-mapping ç»“æœçš„æ•°æ®
        - ğŸ“hg38: å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg38 çš„ fine-mapping ç»“æœçš„æ•°æ®
        - ğŸ“„fine_mapping_result.pkl: ç”¨æ–‡ä»¶å­˜å‚¨æ‰€æœ‰å‚è€ƒåŸºå› ç»„çš„ fine-mapping ç»“æœçš„æ•°æ®
        - ğŸ“„fine_mapping_result_hg19.pkl: ç”¨æ–‡ä»¶å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„ fine-mapping ç»“æœçš„æ•°æ®
        - ğŸ“„fine_mapping_result_hg38.pkl: ç”¨æ–‡ä»¶å­˜å‚¨å‚è€ƒåŸºå› ç»„ä¸º hg19 çš„ fine-mapping ç»“æœçš„æ•°æ®
  - ğŸ“project_code: å­˜å‚¨ä»£ç 
    - ğŸ“scvdb_reproducibility: æ­¤æ–‡ä»¶å¤¹è¡¨ç¤ºæœ¬é¡¹ç›®çš„æ ¹è·¯å¾„
  - ğŸ“scATAC: 
    - ğŸ“`{scATAC-seq}`: 
  - ğŸ“topic: 
  - ğŸ“variant: 



```shell
source_path="/public/home/lcq/rgzn/yuzhengmin/keti/database/sc_variant/table"
target_path="/mnt/data3/workspace/SCVdb"

# scATAC-seq
scp -r "$source_path/download/scatac/*" "root@bio.liclab.net:$target_path/data/data/download/scatac/"

# HOMER
scp -r "$source_path/homer/hg19/*" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/homer/hg19/"
scp -r "$source_path/homer/hg38/*" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/homer/hg38/"
scp -r "$source_path/homer/t_homer_trait_tf.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/homer/"

# MAGMA
scp -r "$source_path/magma/hg19/*" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/magma/hg19/"
scp -r "$source_path/magma/hg19_anno/*" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/magma/hg19_anno/"
scp -r "$source_path/magma/hg38/*" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/magma/hg38/"
scp -r "$source_path/magma/hg38_anno/*" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/magma/hg38_anno/"
scp -r "$source_path/magma/gene_enrichment_trait_table/*" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/magma/gene_enrichment_trait_table/"
scp -r "$source_path/magma/t_magma.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/magma_homer/magma/"

# scATAC-seq
scp -r "$source_path/scatac/difference_gene/*" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/difference_gene"
scp -r "$source_path/scatac/difference_tf/*" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/difference_tf"
scp -r "$source_path/scatac/gene_enrichment_table/*" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/gene_enrichment_table"
scp -r "$source_path/scatac/difference_gene_data.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/"
scp -r "$source_path/scatac/difference_tf_data.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/"
scp -r "$source_path/scatac/sample_gene_data.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/scatac/"

# Overlap
scp -r "$source_path/trait_variant_overlap/gchromvar_sample_enrichment.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/trait_sample/"
scp -r "$source_path/trait_variant_overlap/scavenge_sample_enrichment.txt" "root@bio.liclab.net:$target_path/mysql/mysqlfile/trait_sample/"
scp -r "$source_path/trait_variant_overlap/trs_overlap_gchromvar.h5ad" "root@bio.liclab.net:$target_path/data/data/trait_variant_overlap/"
scp -r "$source_path/trait_variant_overlap/trs_overlap_scavenge.h5ad" "root@bio.liclab.net:$target_path/data/data/trait_variant_overlap/"

# variant
scp -r "$source_path/variant/*" "root@bio.liclab.net:$target_path/mysql/mysqlfile/variant/"

# TRS
scp -r "$source_path/trs/*" "root@bio.liclab.net:$target_path/data/data/trs/"
scp -r "$source_path/trs_big/*" "root@bio.liclab.net:$target_path/data/data/download/trs_big/"


```

